test/mock_sdk.dart

```diff
@@ -160,6 +160,10 @@ class Future<T> {
   factory Future.delayed(Duration duration, [T computation()]) => null;
   factory Future.value([value]) => null;
   static Future wait(List<Future> futures) => null;
+  Future<T> catchError(Function onError, {bool test(Object error)});
+  Future<T> whenComplete(action());
+  Future<T> then(Future<T> onValue(T value), { Function onError });
+  // Original is Future<S> then<S>(FutureOr<S> onValue(T value), { Function onError });
```

Comment:
Because there was nothing, the 3 methods did not appear, so the tests did not pass, so I had to add those 3 methods just like are in the SDK except the 3rd one.

Is there other way to do this?